packages/tailwindcss/src/variants.ts

```diff
@@ -443,7 +443,14 @@ export function createVariants(theme: Theme): Variants {
   variants.compound('not', Compounds.StyleRules | Compounds.AtRules, (ruleNode, variant) => {
     if (variant.variant.kind === 'arbitrary' && variant.variant.relative) return null
 
-    if (variant.modifier) return null
+    if (variant.modifier) {
+      if (variant.variant.kind === 'compound' || variant.variant.kind === 'functional') {
+        variant.variant.modifier = variant.modifier
+        variant.modifier = null
+      } else {
+        return null
+      }
+    }
```

Comment:
It feels super weird to me to mutate the variant we're getting. Could this be handled in parseCandidate or something?

Reply:
Though I guess we'd need some additional metadata for that to work right. I guess as long as this isn't guaranteed to "break" non-compounded versions of the same variant (do we have a test for that?) then it's probably fine.

Reply:
Yeah the mutation breaks stuff:

<img width="1231" alt="Screenshot 2024-11-15 at 14 08 38" src="https://github.com/user-attachments/assets/022f38e4-f455-443b-a07b-fcb4c3b2480d">

Reply:
Handling it during parsing might break things as well, especially when we re-print a candidate.